bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -578,7 +580,20 @@ public static boolean checkIsType(BType sourceType, BType targetType, List<TypeP
                 if (sourceType.getTag() == TypeTags.FINITE_TYPE_TAG) {
                     return isFiniteTypeMatch((BFiniteType) sourceType, targetType);
                 }
-                return TypeTags.isXMLTypeTag(sourceType.getTag());
+                BXMLType target = ((BXMLType) targetType);
```

Comment:
I think we should move the XML logic to `checkIsRecursiveType` now and handle this similar to other constrained types.

Reply:
Moved the logic to `checkIsRecursiveType` method.